visidata/apps/vdsql/_ibis.py

```diff
@@ -80,24 +82,27 @@ def open_vdsql(vd, p, filetype=None):
 class IbisConnectionPool:
     def __init__(self, source, pool=None, total=0):
         self.source = source
-        self.pool = pool if pool is not None else []
-        self.total = total
+        self._local = threading.local()
+        self._lock = Lock()
 
     def __copy__(self):
-        return IbisConnectionPool(self.source, pool=self.pool, total=self.total)
+        return IbisConnectionPool(self.source)
 
     @contextmanager
     def get_conn(self):
-        if not self.pool:
+        if not hasattr(self._local, 'connection'):
             import ibis
-            r = ibis.connect(str(self.source))
-        else:
-            r = self.pool.pop(0)
-
+            with self._lock:
```

Comment:
Do we need this lock if we're just putting things on threading.local() anyway?  Do we have to even have a `_local_` member or can we just use `threading.local()` each time?

Reply:
The examples I found involved the lock, but for our needs it seems to work without it!